### PR TITLE
Retrieve associated Specific Prices on the Product resource

### PR DESCRIPTION
**Pull request by rGaillard**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | This improvement allow to retrieve the associated specific prices on the product resource using the webservices.
| Type?         | improvement 
| Category?     | WS
| BC breaks?    | no
| Deprecations? | no
| How to test?  | 1. Create a specific price for a product<br>2. Retrieve the product using the webservice

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/16426)

### CODE REVIEW

File: classes/Product.php [hunk 1]
```diff
@@ -417,6 +417,7 @@ class ProductCore extends ObjectModel
             'tax_rules_group' => array('type' => self::HAS_ONE),
             'categories' => array('type' => self::HAS_MANY, 'field' => 'id_category', 'object' => 'Category', 'association' => 'category_product'),
             'stock_availables' => array('type' => self::HAS_MANY, 'field' => 'id_stock_available', 'object' => 'StockAvailable', 'association' => 'stock_availables'),
+            'specific_prices' =>    array('type' => self::HAS_MANY, 'field' => 'id_specific_price', 'object' => 'SpecificPrice', 'association' => 'specific_prices'),
```
Comment (PierreRambaud) [inline classes/Product.php]:
```suggestion
            'specific_prices' => array('type' => self::HAS_MANY, 'field' => 'id_specific_price', 'object' => 'SpecificPrice', 'association' => 'specific_prices'),
```

File: classes/Product.php [hunk 2]
```diff
@@ -6260,6 +6269,13 @@ public function getWsManufacturerName()
         return Manufacturer::getNameById((int) $this->id_manufacturer);
     }
 
+	public function getWsSpecificPrices()
```
Comment (PierreRambaud) [inline classes/Product.php]: You mustn't use tabs :)
And could you please add phpdoc?

File: classes/Product.php [hunk 3]
```diff
@@ -6260,6 +6269,13 @@ public function getWsManufacturerName()
...
+	{
+		return Db::getInstance()->executeS('SELECT `id_specific_price` id, `id_product_attribute`
```
Comment (PierreRambaud) [inline classes/Product.php]: The method name says it returns specifc prices, when the sql request return and `id` and an `id_product_attribute` :thinking:

**Review by PierreRambaud — CHANGES_REQUESTED** [review]
You need to run php-cs-fixer :)

File: classes/Product.php (hunk 3 above)
Comment (PierreRambaud) [reply classes/Product.php]: This is why a phpdoc block is required :smile:

File: classes/Product.php (hunk 2 above)
Comment (rGaillard) [reply classes/Product.php]: My editor is configured with tabs and when i run php-cs-fixer on the file with the PrestaShop config it is changing lot of stuff in the file (like the short array syntax) :(

File: classes/Product.php (hunk 3 above)
Comment (rGaillard) [reply classes/Product.php]: I can change the method name to getWsSpecificPricesIds() if you prefer... i named it like this to be consistent with the other webservice getter methods.

**Comment by rGaillard** [comment]
> You need to run php-cs-fixer :)
It is changing a lot of stuff in the file but if it is OK, i will amend the PR.

File: classes/Product.php (hunk 2 above)
Comment (PierreRambaud) [reply classes/Product.php]: Weird, are you sure you use the `.php_cs.dist` configuration?

```
[3.6.8] got@got:~/projects/prestashop/source/develop[develop]$ ./vendor/bin/php-cs-fixer fix classes/Product.php 
Loaded config default from "/home/got/projects/prestashop/source/develop/.php_cs.dist".
Using cache file "/home/got/projects/prestashop/source/develop/var/.php_cs.cache".
Paths from configuration file have been overridden by paths provided as command arguments.

Fixed all files in 0.001 seconds, 12.000 MB memory used
[3.6.8] got@got:~/projects/prestashop/source/develop[develop]$ git status
On branch develop
nothing to commit, working tree clean
```

File: classes/Product.php (hunk 3 above)
Comment (PierreRambaud) [reply classes/Product.php]: If you can yes. The old way to defined method is not obvious, and we can't trust methods name :/

File: classes/Product.php (hunk 2 above)
Comment (rGaillard) [reply classes/Product.php]: OK, not the same php_cs.dist, i was using the file at the prestashop root directory...

Comment (PierreRambaud) [reply classes/Product.php]: yes, this is what we are using too :thinking: 
https://github.com/PrestaShop/PrestaShop/blob/develop/.php_cs.dist

File: classes/Product.php (hunk 3 above)
Comment (rGaillard) [reply classes/Product.php]: Updated

File: classes/Product.php (hunk 2 above)
Comment (rGaillard) [reply classes/Product.php]: Done with the file inside the var instead of root directory.

File: classes/Product.php [hunk 4]
```diff
@@ -6260,6 +6270,13 @@ public function getWsManufacturerName()
         return Manufacturer::getNameById((int) $this->id_manufacturer);
     }
 
+    public function getWsSpecificPricesIds()
+    {
+        return Db::getInstance()->executeS('SELECT `id_specific_price` id, `id_product_attribute`
+		FROM `' . _DB_PREFIX_ . 'specific_price`
+		WHERE `id_product`=' . (int) $this->id);
```
Comment (PierreRambaud) [inline classes/Product.php:6277]:
```php
return Db::getInstance()->executeS(
    'SELECT `id_specific_price` id, `id_product_attribute` ' .
    'FROM `' . _DB_PREFIX_ . 'specific_price` ' .
    'WHERE `id_product`=' . (int) $this->id
);
```

**Comment by Progi1984** [comment]
@rGaillard Some news ?

**Comment by PierreRambaud** [comment]
Hi @rGaillard ,

Since we had no news from you for more than 30 days, I'll close this pull request. Feel free to reopen or open another one if you think it's still relevant.

Thanks!

**Pull request closed without merging**